Client/mods/deathmatch/logic/CClientVehicle.cpp

```diff
@@ -249,8 +249,7 @@ CClientVehicle::~CClientVehicle()
     }
 
     // And the passenger models
-    int i;
-    for (i = 0; i < (sizeof(m_pPassengers) / sizeof(CClientPed*)); i++)
+    for (size_t i = 0; i < (sizeof(m_pPassengers) / sizeof(CClientPed*)); i++)
```

Comment:
How about using a `std::array` instead of `SFixedArray`. `std::array` supports range based for loops and you will be able to replace this construction with `for (const auto& pPassenger : m_pPassengers)`.

Reply:
Or you can add `begin` and `end` methods to `SFixedArray` which enables range based for loops as well.